File: Fase1/FMsport/src/main/java/com/fmSport/services/ReservaService.java
package com.fmSport.services;

import java.time.LocalDate;
import java.util.List;

import com.fmSport.model.Reserva;

public interface ReservaService {

	Reserva crear(Reserva r);
	
	List<Reserva> listar();
	
	Reserva actualizar(Long id,Reserva datos);
	
	boolean borrar(Long id);
	
	public Reserva obtener(Long id);
	
	List<Reserva> porCliente(Long clienteId);
	
	List<Reserva> porSesion(Long sesionId);
	
	Reserva cancelar(Long id);
	
	List<Reserva> porFecha(LocalDate fecha);
}

File: Cliente/FMSport/src/app/services/reserva-service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Reserva } from '../models/reserva';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class ReservaService {
  private baseUrl: string = 'http://localhost:8080/api/reservas';

  constructor(private http: HttpClient) {}

  listar(): Observable<Reserva[]> {
    return this.http.get<Reserva[]>(this.baseUrl);
  }

  obtener(id: number): Observable<Reserva> {
    return this.http.get<Reserva>(`${this.baseUrl}/${id}`);
  }

  crear(r: Reserva): Observable<Reserva> {
    return this.http.post<Reserva>(this.baseUrl, r);
  }

  actualizar(id: number, r: Reserva): Observable<Reserva> {
    return this.http.put<Reserva>(`${this.baseUrl}/${id}`, r);
  }

  borrar(id: number): Observable<void> {
    return this.http.delete<void>(`${this.baseUrl}/${id}`);
  }

  porCliente(clienteId: number): Observable<Reserva[]> {
    return this.http.get<Reserva[]>(`${this.baseUrl}/cliente/${clienteId}`);
  }

  porSesion(sesionId: number): Observable<Reserva[]> {
    return this.http.get<Reserva[]>(`${this.baseUrl}/sesion/${sesionId}`);
  }

  cancelar(id: number): Observable<Reserva> {
    return this.http.put<Reserva>(`${this.baseUrl}/${id}/cancelar`, {});
  }
}
